src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_StringConcat.cs

```diff
@@ -397,6 +706,21 @@ private BoundExpression ConvertConcatExprToString(SyntaxNode syntax, BoundExpres
                 }
             }
 
+            // If expression is of form `constantChar.ToString()` then rewrite it here to a string literal so we can choose better options for lowering later (e.g. fold it with another constant instead of performing concatenation)
+            // NOTE: We request `object.ToString()` as an optional member here to avoid reporting "missing member" diagnostic earlier than we should.
+            // Because if in the end it turns out that we have a simple concatenation of n strings there is no need to report missing `object.ToString()` member
+            if (TryGetSpecialTypeMethod(expr.Syntax, SpecialMember.System_Object__ToString, out MethodSymbol? optionalObjectToString, isOptional: true) &&
+                expr is BoundCall { ReceiverOpt: { Type: NamedTypeSymbol { SpecialType: SpecialType.System_Char } charType, ConstantValueOpt: { IsChar: true } charConstant } } call &&
+                call.Method.GetLeastOverriddenMember(charType) == optionalObjectToString)
```

Comment:
>call.Method.GetLeastOverriddenMember(charType) == optionalObjectToString [](http://example.com/codeflow?start=16&length=72)

Let's suppose `char` doesn't override `ToString`. Is this check going to produce `true` in a case like that? Let's make sure we have a test covering a scenario like that.

Reply:
> Let's suppose `char` doesn't override `ToString`

How can I mock this in a test?

Reply:
> How can I mock this in a test?

I think, there are plenty of tests that create their own flavor of  core library. Let me know if you cannot find one, and I will try find an example. Obviously, we will not be able to execute the scenario and observe how it behaves at runtime, but we should still be able to observe compiler's behavior through some other means, emitted IL, etc.